content/sensu-go/6.8/catalog/catalog-reference.md

```diff
@@ -0,0 +1,1229 @@
+## Private catalogs
+
+The [Sensu Catalog API][2] renders static HTTP API content that the Sensu web UI can consume.
+This means you can create a private enterprise catalog of custom integrations and make it available to users in the Sensu web UI.
+
+Consider forking the official Sensu Catalog repository, https://github.com/sensu/catalog, as a starting point for building your own private catalog.
+
+The Catalog API defines integrations globally rather than by namespace.
+When you create a private catalog, all integrations in your repository are available for all users across namespaces in the web UI.
+
+Read [Build a private catalog of Sensu integrations][17] for more information.
+
+## catalog-api command line interface tool
+
+{{% notice note %}}
+**NOTE**: The catalog-api tool is an alpha feature and may include breaking changes.
+{{% /notice %}}
+
+Sensu's [catalog-api][16] command line interface (CLI) tool uses the [Sensu Catalog API][2] to convert integration files into static API content that you can host on any HTTP web service.
+
+Use the catalog-api tool to [generate a local catalog API][18] for testing as you develop new integrations and to build and run a private catalog.
+Integration files must be stored in a repository that follows the required [organizational framework][16].
+
+The catalog-api tool is written in Go.
+
+### catalog-api subcommands
+
+The catalog-api tool provides the following subcommands:
+
+{{< code text >}}
+catalog-api catalog --help
+USAGE
+  catalog-api catalog [flags] <subcommand> [flags]
+
+SUBCOMMANDS
+  generate  Generate a static catalog API
+  validate  Validate a catalog directory and its integrations
+  server    Serves static catalog API for development purposes
+  preview   Serves static catalog API & preview catalog web application for development purposes
+
+FLAGS
+  -integrations-dir-name integrations  path to the directory containing namespaced integrations
+  -log-level info                      log level of this command ([panic fatal error warn info debug trace])
+  -repo-dir .                          path to the catalog repository
+{{< /code >}}
+
+### Catalog versions
+
+The catalog-api tool generates builds into a checksum-based output directory structure.
+The `version.json` file manages the path to the latest or production catalog API content and instructs the web UI to load catalog contents from the specified checksum directory.
+When you run the command to generate the catalog, catalog-api creates the `version.json` file.
+
+The contents of a `version.json` file are similar to this example:
+
+{{< code text "JSON" >}}
+{
+  "release_sha256": "5029648381dff2426ea247147456b4f1227fd6d9050fa42f0660e67a218f8c87",
+  "last_updated": 1655840571
+}
+{{< /code >}}
+
+If you make any changes to your integration files, the catalog-api tool will generate a new checksum directory.
+To revert to an older build, change the `release_sha256` in `version.json` to point to a different release directory.
```

Comment:
How do users change the release_sha256 in version.json?

Reply:
If a user needs to revert to an older Catalog API release then they can open the version.json file, replace the existing value for the `release_sha256` field with the sha256 of the Catalog API they'd like to serve, and save it.

*Example:*
The current `version.json` file can be found by visiting `https://catalog.sensu.io/version.json`. This file tells the Web UI where to find the latest catalog release.

The current `release_sha256` checksum is `d0e2ba810c3d546c82121406fb5f214e66aeb7fe9026706f9a7391463cf4da19`. The Web UI can now construct the URL to retrieve that release of the catalog: https://catalog.sensu.io/d0e2ba810c3d546c82121406fb5f214e66aeb7fe9026706f9a7391463cf4da19/v1/catalog.json

If a previous release is contained within a directory with a sha256 checksum of `a0070d2c03de050c4ff116a33a675230a38193edac66862c7d2b1526caf91059` then the `release_sha` field in the `version.json` file can be replaced with that checksum.